conjureup/ui/widgets/step.py

```diff
@@ -175,6 +176,19 @@ def generate_additional_input(self):
         self.step_pile.focus_position = self.current_button_index
 
     def submit(self, btn):
+        for i in self.additional_input:
+            if i['input'] and (
+                    i['input'].value is None and self.model.required):
+                self.app.log.debug("Missing default: {}".format(i))
+                self.has_error = True
+                self.step_pile.contents.insert(
+                    1,
+                    (Text(('error_major', 'Missing required input.')),
```

Comment:
Since this gets put in the first position, It'd be cool if the message said which input was missing. I guess just use the i['label'] ?

Now that I think of it, what happens if more than one input is missing? It looks like we'll just show a message about the first one. It'd be great if instead we could collect a list of labels that had missing inputs and then have the message say "Missing required input: foo, bar, baz".